tls/s2n_handshake_io.c

```diff
@@ -489,10 +487,32 @@ int s2n_generate_new_client_session_id(struct s2n_connection *conn)
     return 0;
 }
 
+/* Lets the server flag whether a HelloRetryRequest is needed while processing extensions */
+int s2n_set_hello_retry_required(struct s2n_connection *conn)
+{
+    conn->handshake.handshake_type |= HELLO_RETRY_REQUEST;
+
+    return 0;
+}
+
+/* Lets the server determine whether a HelloRetryRequest should be sent.
+ * A retry is only possible after the first ClientHello (HELLO_RETRY_MSG). */
+bool s2n_is_hello_retry_required(struct s2n_connection *conn)
+{
+    return conn->handshake.handshake_type & HELLO_RETRY_REQUEST;
+    /*return (conn->handshake.message_number <= 1) && (conn->handshake.handshake_type & HELLO_RETRY_REQUEST);*/
```

Comment:
Please remove this instead of commenting it out.

Reply:
Uhg, removed.